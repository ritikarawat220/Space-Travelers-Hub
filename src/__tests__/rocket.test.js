import React from 'react';
import renderer from 'react-test-renderer';
import '@testing-library/jest-dom/extend-expect';
import { render, screen, waitFor } from '@testing-library/react';
import { Provider } from 'react-redux';
import configureMockStore from 'redux-mock-store';
import Rocket from '../components/rocket';

const mockStore = configureMockStore();

describe('RocketList component', () => {
  it('should render loading message when status is loading', () => {
    const store = mockStore({
      rockets: {
        rockets: [],
        status: 'loading',
        error: null,
      },
    });

    render(
      <Provider store={store}>
        <Rocket />
      </Provider>,
    );

    expect(screen.getByText('Loading...')).toBeInTheDocument();
  });

  it('should render error message when status is failed', () => {
    const store = mockStore({
      rockets: {
        rockets: [],
        status: 'failed',
        error: 'Failed to fetch rockets',
      },
    });

    render(
      <Provider store={store}>
        <Rocket />
      </Provider>,
    );

    expect(screen.getByText('Failed to fetch rockets')).toBeInTheDocument();
  });

  it('should render rocket list when status is succeeded', async () => {
    const mockRockets = [
      { id: 'falcon-1', name: 'Falcon 1', description: 'Small launch vehicle' },
      { id: 'falcon-9', name: 'Falcon 9', description: 'Medium lift launch vehicle' },
      { id: 'falcon-heavy', name: 'Falcon Heavy', description: 'Heavy lift launch vehicle' },
    ];

    const store = mockStore({
      rockets: {
        rockets: mockRockets,
        status: 'succeeded',
        error: null,
      },
    });

    render(
      <Provider store={store}>
        <Rocket />
      </Provider>,
    );

    await waitFor(() => {
      expect(screen.getByText('Falcon 1')).toBeInTheDocument();
      expect(screen.getByText('Falcon 9')).toBeInTheDocument();
      expect(screen.getByText('Falcon Heavy')).toBeInTheDocument();
    });
  });

  const store = mockStore({
    rockets: {
      rockets: [],
      status: 'idle',
      error: null,
    },
  });
  const rocket = {
    id: 'falcon1',
    name: 'Falcon 1',
    description: 'Small launch vehicle',
    flickr_images: ['https://www.example.com/falcon1.jpg'],
    reserved: true,
  };

  it('renders correctly when rocket is reserved', () => {
    const tree = renderer
      .create(
        <Provider store={store}>
          <Rocket rocket={rocket} handleReserve={() => {}} onCancel={() => {}} />
        </Provider>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
