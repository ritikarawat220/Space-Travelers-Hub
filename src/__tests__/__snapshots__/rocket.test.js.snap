// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RocketList component renders correctly when rocket is reserved 1`] = `
<section
  className="main"
/>
`;
